In [24]:
import pandas as pd

In [3]:
data = pd.read_csv("Duplicate Customers - Question.csv")


In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2161 entries, 0 to 2160
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   customer_id         2161 non-null   int64 
 1   customer_unique_id  2161 non-null   object
 2   address_line_1      2161 non-null   object
dtypes: int64(1), object(2)
memory usage: 50.8+ KB


In [5]:
data.duplicated().sum()#overall duplicated values in the data

0

In [14]:
data[data.duplicated(subset=['customer_id'])].count()# duplicated values in customer_id column

customer_id           0
customer_unique_id    0
address_line_1        0
dtype: int64

In [13]:
data[data.duplicated(subset=['customer_unique_id'])].count()# duplicated values in customer_unique_id column

customer_id           0
customer_unique_id    0
address_line_1        0
dtype: int64

In [12]:
data[data.duplicated(subset=['address_line_1'])].count()# duplicated values in address_line_1

customer_id           472
customer_unique_id    472
address_line_1        472
dtype: int64

## Explanation
count of every column is 472 because we are filtering out whole row from the data where entries in
address_line_1 are duplicated.

In [17]:
df = data[data.duplicated(subset=['address_line_1'])]

In [19]:
df.to_csv('identified_duplicate_entries.csv', index=False)# getting duplicated entries in csv file

In [20]:
new_data = data.drop_duplicates(subset=['address_line_1'], keep='first')#dropping all the duplicated rows 


In [21]:
new_data

customer_id               customer_unique_id          address_line_1
0         1142810                   AGI Properties      5721 W. 88th Place
1         1142826                    Adkins, Scott      401 West 550 North
2         1142827                Adkins, Scott Sr.           421 Sassafras
3         1142828                    Adomitis.Ryan  551 Hayes Leonard Road
4         1142831                     Ahmed, Jamil    1486 Spyglass Circle
...           ...                              ...                     ...
2150      1181715    James & Marsha Hall (6224184)    3031 Peterson Street
2151      1181717          Monroe, Carol (4665871)     5752 Houston Avenue
2157      1181774  Valpo Meninite Church (9939789)       1305 Silhavy Road
2158      1181775        Codi Rosochacki (2607483)    2951 Glenrose Street
2160      1181777        6445 Satin Ave. (9994909)       6445 Satin Avenue

[1689 rows x 3 columns]

In [23]:
new_data.to_csv('cleaned_data.csv', index=False)#saving cleaned data in csv file